spec/Section 2 -- Language.md

```diff
@@ -980,3 +985,292 @@ and operations.
+that Interface as well as includes all defined fields.
+
+In this example, an Interface type called `Node` is defined:
+
+```graphql
+interface Node {
+  id: ID
+}
+```
+
+#### Union
+
+UnionTypeDefinition : union Name Directives? = UnionMembers
+
+UnionMembers :
+  - NamedType
+  - UnionMembers | NamedType
+
+Union types represent a list of named Object types. Union types are used as the
+type of a field when one of many possible Object types may yielded during
+execution, and no fields are guaranteed. An Object type is a possible type of a
+Union when it is declared by the Union.
+
+In this example, a Union type called `Actor` is defined:
+
+```graphql
+union Actor = User | Business
+```
+
+
+#### Enum
+
+EnumTypeDefinition : enum Name Directives? { EnumValueDefinition+ }
+
+EnumValueDefinition : EnumValue Directives?
+
+EnumValue : Name
+
+Enum types, like Scalar types, also represent leaf values in a GraphQL type
+system. However Enum types describe the set of legal values.
+
+In this example, an Enum type called `Direction` is defined:
+
+```graphql
+enum Direction {
+  NORTH
+  EAST
+  SOUTH
+  WEST
+}
+```
+
+#### Input Object
+
+InputObjectTypeDefinition : input Name Directives? { InputValueDefinition+ }
+
+Input Object types represent complex input values which may be provided as an
+field argument. Input Object types cannot be the return type of an Object or
+Interface field.
+
+In this example, an Input Object called `Point2D` is defined:
+
+```graphql
+input Point2D {
+  x: Float
+  y: Float
+}
+```
+
+### Type Extension
+
+TypeExtensionDefinition : extend ObjectTypeDefinition
+
+Type extensions may be used by client-side tools in order to represent a
+GraphQL type system which has been extended from the type system exposed via
+introspection by a GraphQL service, for example to represent fields which a
+client of GraphQL uses locally, but is not provided by a GraphQL service.
+
+Any fields or interfaces provided by the extension must not already exist on the
+Object type.
+
+In this example, a client only field is added to a `Story` type:
+
+```graphql
+extend type Story {
+  isHiddenLocally: Boolean
+}
+```
+
+### Directive Definition
+
+DirectiveDefinition : directive @ Name ArgumentsDefinition? on DirectiveLocations
+
+DirectiveLocations :
+  - Name
+  - DirectiveLocations | Name
+
+A GraphQL Type System often also includes directives which may be used to
+annotate various nodes in a GraphQL document as an indicator that they should be
+evaluated differently by a validator, executor, or client tool such as a
+code generator.
+
+Since the validation of a GraphQL document includes ensuring that any directives
+used are defined and used correctly, defining a directive allows for a validator
+to be aware of all possible validation rules.
+
+As part of validation, directives are only allowed to be used in locations
+that are explicitly declared. Directive locations must be one of:
+
+ * `QUERY`
+ * `MUTATION`
```

Comment:
I would suggest to add ``` * `SUBSCRIPTION` ``` to this list.
If `SUBSCRIPTION` was omitted intentionally, can someone explain why?

Kind regards :)